metrics.go

```diff
@@ -65,7 +67,8 @@ type Metric struct {
 // Metrics contains the metrics declarations encoded in the metrics.yaml
 // file.
 type Metrics struct {
-	Metrics map[string]Metric
+	Metrics map[string]Metric `yaml:"metrics"`
+	Plan    string            `yaml:"plan"`
```

Comment:
If the only two possibilities are "required" and "optional", why not just have a "Required bool" field ?